internal/scripts/generate-cov-report.sh

```diff
@@ -120,57 +136,40 @@ show_report() {
     eval $CMD
 }
 
-generate_html_report() {
+# Signature: generate_report(OUTPUT_TYPE, RUST_COVERAGE_OUT, CPP_COVERAGE_OUT)
+generate_report() {
     dependency_check grcov
+    dependency_check gcovr
 
-    mkdir -p ./${COVERAGE_DIR}/
-    grcov \
-          **/${LLVM_PROFILE_PATH} \
-          **/**/${LLVM_PROFILE_PATH} \
-          --binary-path ./target/debug \
-          --source-dir . \
-          --output-type html \
-          --branch \
-          --ignore-not-existing \
-          --ignore "**/iceoryx2-cli/*" \
-          --ignore "**/iceoryx2-ffi/*" \
-          --ignore "**/build.rs" \
-          --ignore "**/tests/*" \
-          --ignore "**/examples/*" \
-          --ignore "**/benchmarks/*" \
-          --ignore "**/target/*" \
-          --ignore "**/.cargo/*" \
-          --llvm-path ${LLVM_PATH} \
-          --output-path ./${COVERAGE_DIR}/html
-    sed -i 's/coverage/grcov/' ${COVERAGE_DIR}/html/coverage.json
-    sed -i 's/coverage/grcov/' ${COVERAGE_DIR}/html/badges/*.svg
-}
+    mkdir -p ./${COVERAGE_OUT_DIR}
 
-generate_lcov_report() {
-    dependency_check grcov
-
-    mkdir -p ./${COVERAGE_DIR}/
+    # Generate Coverage files for Rust Code
     grcov \
-          **/${LLVM_PROFILE_PATH} \
-          **/**/${LLVM_PROFILE_PATH} \
-          --binary-path ./target/debug \
-          --source-dir . \
-          --output-type lcov \
-          --branch \
-          --ignore-not-existing \
-          --ignore "**/iceoryx2-cli/*" \
-          --ignore "**/iceoryx2-ffi/*" \
-          --ignore "**/build.rs" \
-          --ignore "**/tests/*" \
-          --ignore "**/examples/*" \
-          --ignore "**/benchmarks/*" \
-          --ignore "**/target/*" \
-          --ignore "**/.cargo/*" \
-          --llvm-path ${LLVM_PATH} \
-          --output-path ./${COVERAGE_DIR}/lcov.info
-}
-
+        **/${LLVM_PROFILE_PATH} \
+        **/**/${LLVM_PROFILE_PATH} \
+        --binary-path target/debug \
+        --source-dir . \
+        --output-type $1 \
```

Comment:
For maintainability, can you add something like this at the top of the function

```
LOCAL_FOO=$1
```

Same for $2 and $3.